federation/pkg/federation-controller/sync/controller.go

```diff
@@ -327,32 +330,90 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 		glog.Errorf("Failed to ensure delete object from underlying clusters finalizer in %s %s: %v",
 			kind, namespacedName, err)
 		s.deliver(namespacedName, 0, false)
-		return
+		return err
 	}
 
 	glog.V(3).Infof("Syncing %s %s in underlying clusters", kind, namespacedName)
+	err, operations := s.getClusterOperations(obj)
+	if err != nil {
+		glog.Errorf("Error getting reconcile cluster operations in %s %s: %v",
+			kind, namespacedName, err)
+		s.deliver(namespacedName, 0, false)
+		return err
+	}
+
+	if len(operations) == 0 {
+		// Everything is in order
+		return nil
+	}
+	err = s.updater.UpdateWithOnError(operations, s.updateTimeout,
+		func(op util.FederatedOperation, operror error) {
+			s.eventRecorder.Eventf(obj, api.EventTypeWarning, "UpdateInClusterFailed",
+				"%s update in cluster %s failed: %v", strings.ToTitle(kind), op.ClusterName, operror)
+		})
+
+	if err != nil {
+		glog.Errorf("Failed to execute updates for %s: %v", key, err)
+		s.deliver(namespacedName, 0, true)
```

Comment:
Why is the last argument true here and false in line 341? Should be same at both places.